ivy/functional/frontends/torch/spectral_ops.py

```diff
@@ -13,11 +12,23 @@ def bartlett_window(
     device=None,
     requires_grad=False
 ):
-    # this implementation is based on scipy.signal.windows.bartlett
-    # https://github.com/scipy/scipy/blob/v1.11.2/scipy/signal/windows/_windows.py#L625-L721
-    if int(window_length) != window_length or window_length < 0:
+    """Generate a Bartlett window.
+
+    Parameters:
+    - window_length (int): Length of the window.
+    - periodic (bool): Whether the window is periodic.
+    - dtype (str, optional): Data type of the output window. Defaults to None.
+    - layout (str, optional): Layout of the output window. Defaults to None.
+    - device (str, optional): Device of the output window. Defaults to None.
+    - requires_grad (bool, optional): Whether the output window requires gradients. Defaults to False.
+
+    Returns:
+    - Ivy array: Bartlett window.
+    """
+    if not isinstance(window_length, int) or window_length < 0:
         raise ValueError("Window length must be a non-negative integer")
-    elif window_length == 1:
+
+    if window_length == 1:
         return ivy.ones(window_length)
```

Comment:
Without testing, it's difficult to determine, but this change could potentially alter the logic of the condition.